demos/elastodynamics/elastodynamics-mwe.py

```diff
@@ -0,0 +1,239 @@
+mu      = E/(2*(1 + nu))
+rho     = 1986.
+
+# The loading parameters are::
+amplitude = 40.0e3
+omega     = 50.
+
+# We create a rectangular column mesh where the number of elements in the
+# :math:`z`-direction is allowed to vary to demonstrate convergence::
+height = 20.
+width  = 1.
+nx     = 1
+ny     = 1
+nz     = 20
+mesh   = BoxMesh(nx, ny, nz, width, width, height)
+
+# Set up outputs::
+u_numerical    = np.zeros((Nsteps_numerical+2))
+time_numerical = np.linspace(0, T, Nsteps_numerical+2)
+
+# We first compute all surface areas of the mesh and then assign
+# the differential area element :math: `\Gamma_N`:: at the top surface
+ds      = Measure('ds', mesh)
+Gamma_N = ds(6)
+
+# We choose a degree 1 continuous vector function space, and set up
+# the function space and functions::
+W = VectorFunctionSpace(mesh, 'Lagrange', 1)
+u = TrialFunction(W)
+w = TestFunction(W)
+
+u_n   = Function(W, name='displacement')
+u_np1 = Function(W)
+u_nm1 = Function(W)
+
+# Next we define the traction vector which will be updated in the time loop::
+traction = Constant((0., 0., 0.))
+
+# We next establish the Dirichlet boundary conditions::
+fixedSides  = Constant(0.)
+fixedBottom = as_vector((0.,0.,0.))
+
+fixedLeft_BC_x  = DirichletBC(W.sub(0), fixedSides, 1)
+fixedLeft_BC_y  = DirichletBC(W.sub(1), fixedSides, 1)
+fixedRight_BC_x = DirichletBC(W.sub(0), fixedSides, 2)
+fixedRight_BC_y = DirichletBC(W.sub(1), fixedSides, 2)
+fixedBack_BC_x  = DirichletBC(W.sub(0), fixedSides, 3)
+fixedBack_BC_y  = DirichletBC(W.sub(1), fixedSides, 3)
+fixedFront_BC_x = DirichletBC(W.sub(0), fixedSides, 4)
+fixedFront_BC_y = DirichletBC(W.sub(1), fixedSides, 4)
+fixedBottom_BC  = DirichletBC(W, fixedBottom, 5)
+
+bcSet = [fixedLeft_BC_x, fixedRight_BC_x, fixedBack_BC_x, fixedFront_BC_x, \
+         fixedLeft_BC_y, fixedRight_BC_y, fixedBack_BC_y, fixedFront_BC_y, \
+         fixedBottom_BC]
+
+print("Computing numerical solution...")
+
+if lump_mass:
+    lumped_mass = assemble((rho/(dt*dt))*dot(w, Function(W).interpolate(as_vector([1,1,1]))) * dx)
+    # Time-stepping loop::
+    while t < T + EPSILON:
+        # Update time step::
+        # print(t)
+        t    += dt
+        step += 1
+        # Update load::
+        traction.assign((0., 0., -0.5*amplitude*(1 - cos(omega*t))))
+        # Assemble right-hand side::
+        rhs = assemble(-inner(sigma(u_np1), epsilon(w))*dx + dot(traction, w)*Gamma_N)
+        us  = Function(W).assign(rhs)
+        # Divide by lumped mass matrix::
+        us /= lumped_mass
+        # Update displacements::
+        u_np1.assign(us + 2*u_n - u_nm1)
+        for bc in bcSet:
+            bc.apply(u_np1)
+        u_n.assign(u_np1)
+        u_nm1.assign(u_n)
+        # Save data::
+        u_numerical[step] = u_n([0., 0., 20.])[2]
+
+else:
+    # Apply the central differencing scheme::
+    F    = inner(sigma(u), epsilon(w))*dx + rho/(dt*dt)*dot(u - 2*u_n + u_nm1, w)*dx - dot(traction, w)*Gamma_N
```

Comment:
```
F = inner(sigma(u_n), epilon(w)) * dx + ...
```
I think. I *think* this way we can make sure that the initial velocity (which I assume is `0`) is approximated to second-order.